factory/usr/lib/core/extra-paths

```diff
@@ -2,8 +2,14 @@
 
 set -eu
 
-cat <<EOF >>/sysroot/etc/fstab
+cat <<'EOF' >>/sysroot/etc/fstab
 /run/mnt/data /writable none bind,x-initrd.mount 0 0
+EOF
+
+# Mount /lib/{firmware,modules} only if there is no drivers tree created by snapd
+if [ ! -d /run/mnt/data/system-data/var/lib/snapd/kernel ]; then
```

Comment:
What are the situation where this is not created? I think of 2, but I want to make sure I understand.
* When snapd is too old (will not be the case in uc26 anymore)
* It is the first boot, or factory reset

Reply:
Could happen for old snapd or base < core24 so snapd does not create the drivers tree. Also on install mode or factory reset, when you have an ephemeral system. But for first boot after installation the drivers tree should be there (see https://github.com/snapcore/snapd/pull/13923).

Reply:
No, we do not generate that mount in `/etc`.

In any case, as discussed it would be good to check for the path including the revision (`/var/lib/snapd/kernel/<kernel_snap_name>/<snap_revision>/`)

Reply:
Modules are mounted by the systemd generator (https://github.com/snapcore/snapd/blob/master/cmd/snapd-generator/main.c) that now we include in core24.